dask/array/core.py

```diff
@@ -2750,24 +2751,30 @@ def elemwise(op, *args, **kwargs):
         # their values to compute the result dtype:
         # https://github.com/numpy/numpy/issues/6240
         # We don't inspect the values of 0d dask arrays, because these could
-        # hold potentially very expensive calculations.
-        vals = [np.empty((1,) * a.ndim, dtype=a.dtype)
+        # hold potentially very expensive calculations. Instead, we treat
+        # them just like other arrays, and if necessary cast the result of op
+        # to match.
+        vals = [np.ones((1,) * max(1, a.ndim), dtype=a.dtype)
```

Comment:
I don't think we need to switch to `ones` at all. The issue was the shape, not the value. But `ones` does work for record dtypes.

```python
In [2]: np.ones(3, dtype=[('a', 'i8'), ('b', 'S2')])
Out[2]:
array([(1, b'1'), (1, b'1'), (1, b'1')],
      dtype=[('a', '<i8'), ('b', 'S2')])
```